docs/2.guide/4.recipes/4.sessions-and-authentication.md

```diff
@@ -0,0 +1,403 @@
+  if (!foundUser || !bcrypt.compareSync(password, foundUser.password)) {
+    // return an error if the user is not found or the password doesn't match
+    throw createError({
+      statusCode: 401,
+      statusMessage: "Invalid email or password",
+    });
+  }
+
+  // log in as the selected user
+  await replaceUserSession(event, {
+    user: {
+      id: user.id,
+      name: user.name,
+    },
+    loggedInAt: new Date(),
+  });
+});
+
+
+```
+
+The user should now be logged in! With the session set, we can get the current user session in any API route or page by calling `getUserSession(event)` which is auto-imported as a util function from the `nuxt-auth-utils` package.
+
+### 6. Create a logout API route
+
+Users need to be able to log out, so we should create an API route to allow them to do this. This should require post request as well, just like login. We'll clear the session when this endpoint is called.
+
+We'll use the `clearUserSession` from the `auth-utils` module to log the user out and clear the session data. This function is automatically imported from the module by Nuxt, so we don't need to manually import it.
+
+```typescript [server/api/auth/logout.post.ts]
+export default defineEventHandler(async (event) => {
+  // Clear the current user session
+  await clearUserSession(event);
+});
+```
+
+### 7. Protect your server route
+
+Protecting server routes is key to making sure your data are safe. Front-end middleware is helpful for the user, but without back-end protection your data can still be accessed. Because of this, it is critical that we protect any API routes with sensitive data. For these sensitive routes, we should return a 401 error if the user is not logged in.
+
+The `auth-utils` module provides the `requireUserSession` utility function to help make sure that users are logged in and have an active session. We can use this to protect our different endpoints. Like many of the other utilities from the auth module, it is automatically imported in our server endpoints.
+
+In the example below, we use the `requireUserSession` utility function to protect the `/server/api/users.get.ts` server route. This route will only be accessible to logged-in users.
+
+```typescript [/server/api/users.get.ts]
+import getDatabase from "~/database/database";
+import users from "~/database/schema/users";
+import requireUserLoggedIn from "~/server/utils/requireUserLoggedIn";
+
+export default defineEventHandler(async (event) => {
+  // make sure the user is logged in
+  // This will throw a 401 error if the request doesn't come from a valid user session
+  await requireUserSession(event);
+
+  // If we make it hear, the user is authenticated. It's safe to fetch and return data
+  const db = await getDatabase();
+  
+  // Send back the list of users
+  const userList = await db.select({ name: users.name, id: users.id }).from(users).limit(10);
+
+  return userList;
+});
+```
+
+### 8. Create a front-end middleware to protect routes
```

Comment:
Here too we should consider using the more friendly client-side term in favor of front-end

Reply:
I've replaced "front-end" with "client-side" and "back-end" with "server-side" throughout the document.

Reply:
Sounds good! The less technical terms the more beginner friendly it would be